networking/eth2/src/main/java/tech/pegasys/artemis/networking/eth2/gossip/topics/AttestationTopicHandler.java

```diff
@@ -27,29 +27,29 @@
 import tech.pegasys.artemis.datastructures.operations.IndexedAttestation;
 import tech.pegasys.artemis.datastructures.state.BeaconState;
 import tech.pegasys.artemis.datastructures.util.SimpleOffsetSerializer;
+import tech.pegasys.artemis.ssz.SSZTypes.Bytes4;
 import tech.pegasys.artemis.storage.client.RecentChainData;
 
 public class AttestationTopicHandler extends Eth2TopicHandler<Attestation> {
 
   private static final Logger LOG = LogManager.getLogger();
 
-  private final String attestationsTopic;
   private final RecentChainData recentChainData;
+  private final UnsignedLong subnetId;
 
   public AttestationTopicHandler(
-      final EventBus eventBus, final RecentChainData recentChainData, final UnsignedLong subnetId) {
-    super(eventBus);
-    this.attestationsTopic = getTopic(subnetId);
+      final EventBus eventBus,
+      final RecentChainData recentChainData,
+      final UnsignedLong subnetId,
+      final Bytes4 forkDigest) {
+    super(eventBus, forkDigest);
     this.recentChainData = recentChainData;
-  }
-
-  private static String getTopic(final UnsignedLong subnetId) {
-    return "/eth2/committee_index" + toIntExact(subnetId.longValue()) + "_beacon_attestation/ssz";
+    this.subnetId = subnetId;
   }
 
   @Override
-  public String getTopic() {
-    return attestationsTopic;
+  public String getTopicName() {
+    return "committee_index" + toIntExact(subnetId.longValue()) + "_beacon_attestation";
```

Comment:
We should probably validate that the subnetId int is positive

Reply:
It's an UnsignedLong?

Reply:
UnsignedLong.MAX_VALUE.longValue() == -1

Reply:
In most parts of the codebase if we were dealing with values from 2^63 to 2^64-1 we would have the same issue but I don't think it's in the scope of this PR.

Reply:
It's a good point though, so I made a ticket: https://pegasys1.atlassian.net/browse/BC-397